### PR TITLE
Add `meta_state` testing

### PR DESCRIPTION
**Pull request by fluiderson**
* Updated tests for the `meta_state` testing feature from gtest (https://github.com/gear-tech/gear/pull/944).
* Refactored and added a few new docs.
* Added restriction for the zero address among producers, distributors, and retailers in initialization config.

### CODE REVIEW

File: io/src/lib.rs [hunk 1]
```diff
@@ -80,12 +92,13 @@ pub enum SupplyChainAction {
...
-    /// * `item_id`: an item's ID.
-    /// * `approve`: yes ([`true`]) or no ([`false`]).
-    ///
-    /// On success returns [`SupplyChainEvent::Success`].
-    ApproveByProducer { item_id: ItemId, approve: bool },
+    /// On success, returns [`SupplyChainEvent::Success`].
+    ApproveByProducer {
+        /// An item's ID.
+        item_id: ItemId,
+        /// Yes ([`true`]) or no ([`false`]).
```
Comment (shamilsan) [inline io/src/lib.rs:99]: No need for links to bool values I guess. Example: https://doc.rust-lang.org/src/core/option.rs.html#538
```suggestion
        /// Yes (`true`) or no (`false`).
```

File: io/src/lib.rs [hunk 2]
```diff
@@ -182,12 +209,13 @@ pub enum SupplyChainAction {
...
-    /// * `item_id`: an item's ID.
-    /// * `approve`: yes ([`true`]) or no ([`false`]).
-    ///
-    /// On success returns [`SupplyChainEvent::Success`].
-    ApproveByDistributor { item_id: ItemId, approve: bool },
+    /// On success, returns [`SupplyChainEvent::Success`].
+    ApproveByDistributor {
+        /// An item's ID.
+        item_id: ItemId,
+        /// Yes ([`true`]) or no ([`false`]).
```
Comment (shamilsan) [inline io/src/lib.rs:216]:
```suggestion
        /// Yes (`true`) or no (`false`).
```

File: src/lib.rs [hunk 3]
```diff
@@ -16,30 +18,30 @@ fn panic_item_not_exist(item_id: ItemId) -> ! {
     panic!("Item with the {item_id} ID doesn't exist")
 }
 
+fn panic_zero_address_among_(who: &str) -> ! {
```
Comment (shamilsan) [inline src/lib.rs:21]: Why the underscore at the end of the function name?
```suggestion
fn panic_zero_address_among(who: &str) -> ! {
```

**Review by shamilsan — APPROVED** [review]
Great 🔥

File: io/src/lib.rs (hunk 1 above)
Comment (shamilsan) [reply io/src/lib.rs:99]: Ah, somebody needs to check what does `true` or `false` mean? 😄  Ok if so.

**Review by LouiseMedova — APPROVED** [review]

**Review by shamilsan — APPROVED** [review]

**Pull request merged**